dali/operators/generic/slice/slice_attr.h

```diff
@@ -33,13 +35,30 @@
 
 namespace dali {
 
+namespace detail {
+
+enum class AxesMode : uint8_t {
+  Axes, AxisNames, AllAxes
+};
+
+/**
+ * @brief Gets a small vector with the final axes for a particular sample, given the current config
+ */
+SmallVector<int, 8> GetAxes(AxesMode axes_mode, const TensorLayout& shape_layout,
+                            const TensorLayout& axis_names, const ArgValue<int, 1>& axes_arg,
+                            int data_idx, int ndim);
+
+
 /**
  * @brief Process axes arguments.
  * @return true if there are axes defined, false otherwise (use all axes).
  */
-bool ProcessAxesArgs(std::vector<int>& axes, TensorLayout& axis_names, const OpSpec& spec,
-                     const char* axes_arg_name = "axes",
-                     const char* axis_names_arg_name = "axis_names");
+AxesMode ProcessAxesArgs(TensorLayout& axis_names, const OpSpec& spec,
```

Comment:
Please update the `@return` description. It is not `bool` anymore.

Reply:
Done